dotfile/src/lib.rs

```diff
@@ -29,13 +36,82 @@ impl SetupAdapter for DotfileSetup {
         &self,
         config_string: &str,
         config_extension: &str,
-        _base_config: &BasicConfigContent,
+        base_config: &BasicConfigContent,
     ) -> Result<()> {
         let config: DotfileConfiguration =
             core::configuration::generate_config(config_string, config_extension)?;
 
-        println!("{config:?}");
+        config.create_files(base_config);
+        config.create_program_files(base_config)?;
+
+        Ok(())
+    }
+}
+
+impl DotfileConfiguration {
+    fn create_files(&self, base_config: &BasicConfigContent) {
+        self.files
+            .iter()
+            .flatten()
+            .for_each(|f| f.create_system_file(base_config));
+    }
+
+    fn create_program_files(&self, base_config: &BasicConfigContent) -> Result<()> {
+        let nvim = match &self.nvim {
+            Some(nvim) => nvim,
+            None => return Ok(()),
+        };
+
+        let target_path = format!(
+            "{}/{}",
+            &base_config
+                .base
+                .base_path
+                .to_str()
+                .expect("this should have already been validated to exist."),
+            GitModule::get_project_path(nvim, base_config)
+                .expect("Malformed git resource mentioned!"),
+        );
+
+        GitModule::git_clone(nvim, &target_path)?;
 
         Ok(())
     }
 }
+
+impl DotfileDefinition {
+    fn create_system_file(&self, base_config: &BasicConfigContent) {
+        if let Some(parent) = self.path.parent() {
+            let _ = fs::create_dir_all(parent);
+        }
```

Comment:
Block feels awkward. Swallowing an error, fighting Option. After converting this function to return result try something like this:

```suggestion
        self.path.parent().map(fs::create_dir_all)?;
```